src/config.rs

```diff
@@ -1285,41 +1286,39 @@ pub mod config_file {
     /// Returns the parsed `Some(ConfigFile)` if the config file exists, `None` otherwise.
     ///
     /// Trippy will attempt to locate a `trippy.toml` or `.trippy.toml`
-    /// config file in one of the following platform specific locations:
+    /// config file in one of the following locations:
     ///     - the current directory
     ///     - the user home directory
-    ///     - the user config direction
-    ///
-    /// For example, on Linux the Trippy will attempt to locate the following
-    /// files (in order):
-    ///     - `./trippy.toml`
-    ///     - `./.trippy.toml`
-    ///     - `$HOME/trippy.toml`
-    ///     - `$HOME/.trippy.toml`
-    ///     - `$HOME/.config/trippy.toml`
-    ///     - `$HOME/.config/.trippy.toml`
-    ///
-    /// See [here](https://github.com/dirs-dev/dirs-rs) for platform specific directory
-    /// information.
+    ///     - the XDG config directory (Unix only): `$XDG_CONFIG_HOME` or `~/.config`
+    ///     - the macOS data directory (macOS only): `~/Library/Application Support`
+    ///     - the Windows data directory (Windows only): `%APPDATA%`
     ///
     /// Note that only the first config file found is used, no attempt is
     /// made to merge the values from multiple files.
     pub fn read_default_config_file() -> anyhow::Result<Option<ConfigFile>> {
-        if let Some(file) = read_file(|| Some(PathBuf::new()), DEFAULT_CONFIG_FILE)? {
-            Ok(Some(file))
-        } else if let Some(file) = read_file(|| Some(PathBuf::new()), DEFAULT_HIDDEN_CONFIG_FILE)? {
-            Ok(Some(file))
-        } else if let Some(file) = read_file(dirs::home_dir, DEFAULT_CONFIG_FILE)? {
-            Ok(Some(file))
-        } else if let Some(file) = read_file(dirs::home_dir, DEFAULT_HIDDEN_CONFIG_FILE)? {
-            Ok(Some(file))
-        } else if let Some(file) = read_file(dirs::config_dir, DEFAULT_CONFIG_FILE)? {
-            Ok(Some(file))
-        } else if let Some(file) = read_file(dirs::config_dir, DEFAULT_HIDDEN_CONFIG_FILE)? {
-            Ok(Some(file))
-        } else {
-            Ok(None)
+        if let Some(file) = read_file("")? {
+            return Ok(Some(file));
+        }
+
+        let home_dir = etcetera::home_dir()?;
+        if let Some(file) = read_file(home_dir)? {
+            return Ok(Some(file));
+        }
+
+        let config_dir = etcetera::base_strategy::choose_base_strategy()?.config_dir();
+        if let Some(file) = read_file(config_dir)? {
+            return Ok(Some(file));
         }
+
+        #[cfg(target_os = "macos")]
```

Comment:
Not a fan of platform specific cfg in the application layer.  We could just check for the macOS dir on all platforms but in the end I think we can just ignore the official macOS config dir and stick to XDG.

Reply:
Yeah, I don't think anyone really wants to use the Library path. I only included it because I thought you might not want a breaking change.

Reply:
Good thinking, luckily the config file is new in `0.8.0` which is unreleased so there is no breaking change here.